<a href="https://colab.research.google.com/github/nurfnick/Numerical_Methods/blob/master/WeBWorK2HomeworkHelper.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# WeBWorK Homework Helper 2



## Number 1

The definition of the derivative, applied to $f(x) = 7^x$, results in the formula $f'(x) = k \cdot 7^x$ where
$$k = \lim_{h \rightarrow 0} \frac{7^h - 1}{h}$$ .

The value of the constant $k$ cannot be determined using the usual tricks for evaluating limits. Using your calculator, approximate the constant $k$, rounded to three significant figures.

In [2]:
def quo(h):
  return (7**h-1)/h

testvalues =[1,.1,.01,.001,.0001]

for i in testvalues:
  print(quo(i))

6.0
2.148140440390669
1.9649663856459165
1.9478046608600863
1.9460994896514094


I want three sig figs, so I think the answer is 1.95 if I round up.  If I needed another sig fig, I would go ahead and try another lower power of ten.

## WeBWorK 11

The temperature adjusted for wind chill is a temperature that tells you how cold it feels, as a result of the combination of wind speed, w, in miles per hour, and actual temperature, T, as measured by a thermometer in degrees Fahrenheit. The table below shows the temperature adjusted for wind chill, f(w,T), as a function of w and T. The left column of the table gives the wind speed in miles per hour, the top row gives the actual temperature in degrees Fahrenheit, and the remaining entries give the temperature adjusted for wind chill in degrees Fahrenheit as a function of wind speed and actual temperature. 

| w\T | 25 | 20 | 15 | 10  | 5   | 0   | -5  | -10 |
|-----|----|----|----|-----|-----|-----|-----|-----|
| 5   | 19 | 13 | 7  | 1   | -5  | -11 | -16 | -22 |
| 10  | 15 | 9  | 3  | -4  | -10 | -16 | -22 | -28 |
| 15  | 13 | 6  | 0  | -7  | -13 | -19 | -26 | -32 |
| 20  | 11 | 4  | -2 | -9  | -15 | -22 | -29 | -35 |
| 25  | 9  | 3  | -4 | -11 | -17 | -24 | -31 | -37 |

 Estimate $f_w(10,10)$ and give a practical interpretation. $f_w(10,10)\approx$

If the wind is blowing mph and the actual temperature is degrees Fahrenheit, it will feel degrees Fahrenheit colder for each 1 mph increase in wind speed.


So I know that many of you have not taken multivariate calculus but the real world does not just have one variable!  To estimate partial derivatives, we will do the same as with one variable and holding the other values constant.  There are actually two estiates we can make here and I'll make them both.

$$f_w(10,10)\approx \frac{f(10+h,10)-f(10,10)}h
  $$

I can use $h=\pm5$

In [1]:
# h is +5

print((-7+4)/5)

# h is -5

print( (1+4)/(-5))

-0.6
-1.0


## WeBWorK 23

Find a polynomial of the form
$$f(x)=ax^3+bx^2+cx+d$$
such that $f(0)=5$, $f(2)=6$, $f(3)=−6$, and $f(−4)=7$. 

I'll apply Lagrange Interpolation.

In [3]:
def lagrangeCoefficient(inputArray,outputArray,entry,variable):
  lc = 1
  for i in range(len(inputArray)):
    if i == entry:
      continue
    else:
      lc = lc*(variable - inputArray[i])/(inputArray[entry]-inputArray[i])
  return lc

def lagrangeInterpolator(inputArray, outputArray, variable):
  li = 0
  for i in range(len(inputArray)):
    li = li + lagrangeCoefficient(inputArray,outputArray,i,variable)*outputArray[i]
  return li

In [4]:
input = [0,2,3,-4]
output = [5,6,-6,7]

lagrangeInterpolator(input,output,0) #this is d


5.0

Again though the issue I am going to have is that I cannot do this symbolically...  I can give you the numerical answer here (give me an $x$ I'll tell you the output) but this is not what is wanted.

$$
f(x) = y_1\frac{(x-x_2)\cdots (x-x_n)}{(x_1-x_2)\cdots (x_1-x_n)}+\cdots +y_n\frac{(x-x_1)\cdots (x-x_{n-1})}{(x_n-x_1)\cdots (x_n-x_{n-1})}
$$

So here the answer is

$$
f(x) =5\frac{(x-2)(x-3)(x+4)}{(-2)(-3)(4)}+6\frac{x(x-3)(x+4)}{2(-1)(6)}-6\frac{x(x-2)(x+4)}{x(x-2)(x+4)}+7\frac{x(x-2)(x-3)}{-4(-2)(-1)}
$$

I don't find this very satisfying but I did it!